### PR TITLE
Delete authToken from MSAppCenterIngestion and pass it for each batch

### PR DESCRIPTION
**Pull request by roland-chernov-akvelon**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

This PR deletes `authToken` from `MSAppCenterIngestion` and passes it for each batch to avoid sending incorrect token when token is changed.

### CODE REVIEW

File: AppCenter/AppCenter/Internals/Ingestion/MSHttpIngestion.h [hunk 1]
```diff
@@ -40,11 +40,13 @@ static NSString *const kMSETagRequestHeader = @"If-None-Match";
  *
  * @param data A data instance that will be transformed request body.
  * @param eTag HTTP entity tag.
+ * @param authToken Auth token to send data with
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Ingestion/MSHttpIngestion.h]: Dot at the end

File: AppCenter/AppCenterTests/MSOneCollectorIngestionTests.m [hunk 2]
```diff
@@ -285,11 +288,12 @@ - (void)testCompressHTTPBody {
   OCMStub([deviceMock isValid]).andReturn(YES);
   MSMockCommonSchemaLog *log1 = [[MSMockCommonSchemaLog alloc] init];
   log1.sid = @"";
+  log1.name = @"";
   log1.timestamp = [NSDate date];
   log1.ext = [MSModelTestsUtililty extensionsWithDummyValues:[MSModelTestsUtililty extensionDummies]];
   NSMutableString *jsonString = [NSMutableString new];
   log1.sid = [log1.sid stringByPaddingToLength:kMSHTTPMinGZipLength withString:@"." startingAtIndex:0];
-  log1.name = [log1.sid stringByPaddingToLength:kMSHTTPMinGZipLength withString:@"abcd" startingAtIndex:0];
+  log1.name = [log1.name stringByPaddingToLength:kMSHTTPMinGZipLength withString:@"abcd" startingAtIndex:0];
```
Comment (MatkovIvan) [inline AppCenter/AppCenterTests/MSOneCollectorIngestionTests.m]: 👍

File: AppCenter/AppCenter/Internals/Ingestion/MSIngestionProtocol.h [hunk 3]
```diff
@@ -26,6 +26,17 @@ NS_ASSUME_NONNULL_BEGIN
...
 @property(nonatomic, readonly, getter=isReadyToSend) BOOL readyToSend;
 
+/**
+ * Send data.
+ *
+ * @param data Instance that will be transformed to request body.
+ * @param authToken Auth token to send data with.
+ * @param handler Completion handler.
+ */
+- (void)sendAsync:(nullable NSObject *)data
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/Internals/Ingestion/MSIngestionProtocol.h]: This is going to be a break change in `feature/storage` branch. Please also consider to refactor classes inherited from this  class.

File: AppCenter/AppCenter/Internals/Ingestion/MSIngestionProtocol.h [hunk 4]
```diff
@@ -34,6 +45,19 @@ NS_ASSUME_NONNULL_BEGIN
...
 
+/**
+ * Send data.
+ *
+ * @param data Instance that will be transformed to request body.
+ * @param eTag HTTP entity tag.
+ * @param authToken Auth token to send data with.
+ * @param handler Completion handler.
+ */
+- (void)sendAsync:(nullable NSObject *)data
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/Internals/Ingestion/MSIngestionProtocol.h]: This one too.

File: AppCenter/AppCenter/Internals/Channel/MSChannelGroupDefaultPrivate.h [hunk 5]
```diff
@@ -5,7 +5,7 @@ NS_ASSUME_NONNULL_BEGIN
 
 @class MSAppCenterIngestion;
 
-@interface MSChannelGroupDefault () <MSAuthTokenContextDelegate, MSChannelDelegate>
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/Internals/Channel/MSChannelGroupDefaultPrivate.h]: Is `MSAuthTokenContextDelegate` still being used?

Comment (MatkovIvan) [reply AppCenter/AppCenter/Internals/Channel/MSChannelGroupDefaultPrivate.h]: @jaelim-ms yes, at least in `feature/storage` branch.

File: AppCenter/AppCenter/Internals/Ingestion/MSIngestionProtocol.h (hunk 3 above)
Comment (roland-chernov-akvelon) [reply AppCenter/AppCenter/Internals/Ingestion/MSIngestionProtocol.h]: Good point! We need to create a separate PR against `feature/storage` branch. For tracking we will create a separate user story to merge `feature/identity` to `feature/storage`, to resolve conflicts and breaking changes.

**Pull request merged**